### PR TITLE
[ftgl] Fix build failure #9520

### PR DESCRIPTION
**Pull request by sma-github**
- Fixes issue #9520 
- Tested on ubuntu 18.04.1
- Solution is the same as cbezault did for issue #7697 to fix the creation of [geographiclib](https://github.com/microsoft/vcpkg/pull/7697/files/89d8cece0dbbfab416a955289bf29b33b0d106b0)

### CODE REVIEW

File: ports/ftgl/portfile.cmake [hunk 1]
```diff
@@ -16,7 +16,13 @@ vcpkg_configure_cmake(
...
 
 vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets(CONFIG_PATH cmake)
+if (VCPKG_TARGET_IS_WINDOWS)
+        vcpkg_fixup_cmake_targets(CONFIG_PATH cmake)
+elseif (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake)
+elseif (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+        vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake)
+endif()
```
Comment (NancyLi1013) [inline ports/ftgl/portfile.cmake]: `VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux"` and `VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin"` may be replaced by `VCPKG_TARGET_IS_LINUX` and `VCPKG_TARGET_IS_OSX`.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/ftgl/portfile.cmake (hunk 1 above)
Comment (sma-github) [reply ports/ftgl/portfile.cmake]: All sugesstions done with [commit 701dacc54...](https://github.com/microsoft/vcpkg/pull/9521/commits/701dacc54b3573eb12503aacbe4169798e3d6ed3)

**Review by NancyLi1013 — APPROVED** [review]

**Comment by dan-shaw** [comment]
Can you update the baseline? It is located at `vcpkg/ci.baseline.txt`. Currently, it says `ftgl:x64-linux=fail`. Since it is passing, we need to remove this line.

**Comment by NancyLi1013** [comment]
Could you solve the conflicts in this PR?

**Comment by NancyLi1013** [comment]
Hi @sma-github thanks for this PR.
It failed on Linux platform due to this:
```
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:295 (message):
  Could not find doxygen.  Please install it via your package manager.
Call Stack (most recent call first):
  ports/ftgl/portfile.cmake:1 (vcpkg_find_acquire_program)
  scripts/ports.cmake:94 (include)
```
Since `doxygen `is only for windows now. So we need to install it manually on other platform.
I test it on my local. Even though I have installed `doxygen` , it still build failed.
```
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^
In file included from /home/linux001/9521/vcpkg/buildtrees/ftgl/src/de901d4e74-1e9729eccc/src/FTGL/ftgl.h:137:0,
                 from /home/linux001/9521/vcpkg/buildtrees/ftgl/src/de901d4e74-1e9729eccc/src/FTBuffer.cpp:28:
/home/linux001/9521/vcpkg/buildtrees/ftgl/src/de901d4e74-1e9729eccc/src/FTGL/FTLibrary.h:137:14: error: ‘atomic’ in namespace ‘std’ does not name a template type
         std::atomic <int> LegacyOpenGLStateHandling;
              ^
/home/linux001/9521/vcpkg/buildtrees/ftgl/src/de901d4e74-1e9729eccc/src/FTGL/FTLibrary.h: In member function ‘bool FTLibrary::GetLegacyOpenGLStateSet() const’:
/home/linux001/9521/vcpkg/buildtrees/ftgl/src/de901d4e74-1e9729eccc/src/FTGL/FTLibrary.h:97:55: error: ‘LegacyOpenGLStateHandling’ was not declared in this scope
         bool GetLegacyOpenGLStateSet() const { return LegacyOpenGLStateHandling; }
```
It seems that this port still doesn't support Linux now.

**Comment by sma-github** [comment]
Hi NancyLi1013,

there seem to be two issues.
- The Doxygen target seems to be a manual target, not called automatically during the build (tested with and without ninja-build here). If you say "doxygen is only for windows now", I could disable the relevant parts with a patch for the non-windows platforms.

- Analyzing your logs, there seems to be problem with the compiler and the supported C++ standard. Which compiler (in which version) is used on the test system? I've using g++ 7.4.0 and have no issues. This [page](https://gcc.gnu.org/gcc-6/porting_to.html), says that from gcc6 the default standard is now GNU++14.  Perhaps, adding a property to the CMake targets that explicitly requesting the flags for the C++11 could help.

What do you think?

**Comment by NancyLi1013** [comment]
@sma-github thanks for your detail analysis.
As for `doxygen`, maybe we can update as the follows:
```
if (VCPKG_TARGET_IS_WINDOWS)
    vcpkg_find_acquire_program(DOXYGEN)
else()
    # doxygen only have windows package in vcpkg now.
    if (NOT DOXYGEN)
        message(FATAL_ERROR "${PORT} requires doxygen, these can be installed on Ubuntu systems via apt-get install doxygen.")
    endif()
endif()
```
For another issue I posted above, I used gcc 5.0.0 and gcc 8.3.0 to test. This issue only appeared on the test machine with gcc 5.0.0.
So in order to make sure this port can be installed successfully on different gcc versions. We need to add `set (CMAKE_CXX_STANDARD 11)`

Please let me know if you have any questions about this.

**Comment by sma-github** [comment]
vcpkg-Linux-PR-test is passing now, but vcpkg-osx-PR-test  fails now.

> ##[error] PASSING, REMOVE FROM FAIL LIST: ftgl:x64-osx (/Users/vagrant/azure-agent/_work/3/s/scripts/ci.baseline.txt)

Should I remove the line "ftgl:x64-osx" from ci.baseline.txt as well? I don't have an osx system to really verify if it works. Or should I set it to "_skip_"?

**Comment by NancyLi1013** [comment]
Since it has passed on osx pipeline.
We need to remove `ftgl:x64-osx=fai`l from ci.baseline.txt.

**Review by JackBoosY — APPROVED** [review]

**Comment by sma-github** [comment]
I've added a missing patch (be7f217), after re-basing and correct the git history. All checks passed now and no additional commits are planned for this pull request from my side.

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**